provisionersdk/provisionertags.go

```diff
@@ -39,3 +41,23 @@ func MutateTags(userID uuid.UUID, provided map[string]string) map[string]string
 	}
 	return tags
 }
+
+// mergeTags merges multiple sets of provisioner tags.
+// Given maps a and b, the result will contain all
+// keys and values contained in a and in b.
+// An exception is made for key-value pairs for which
+// a[key] is non-empty but b[key] is empty. In this case
+// the non-empty value from a will win.
+func mergeTags(a, b map[string]string) map[string]string {
+	m := make(map[string]string)
```

Comment:
Is it possible to receive nils via `a` or `b`? We may consider:

```go
if a == nil {
    a = make(map[string]string)
}
if b == nil {
   b = make(map[string]string)
}
```

Reply:
Ranging over a nil map is fine though?